### PR TITLE
feat: lapsed user prompt

### PR DESCRIPTION
**Pull request by jonathanyeung**
Prompt users with a modal dialog to get started with Dendron under the following conditions:

1) Have not initialized a workspace before
2A) >1 week since install
2B) >1 week since last reminder prompt.

### CODE REVIEW

File: packages/plugin-core/src/_extension.ts [hunk 1]
```diff
@@ -522,4 +524,55 @@ async function showWelcomeOrWhatsNew({
...
+function showLapsedUserMessage() {
+  const START_TITLE = "Get Started";
+
+  AnalyticsUtils.track(VSCodeEvents.ShowLapsedUserMessage);
+  MetadataService.instance().setLapsedUserMsgSendTime();
+
+  vscode.window
+    .showInformationMessage(
+      "Get started with Dendron.",
+      { modal: true },
```
Comment (kevinslin) [inline packages/plugin-core/src/_extension.ts:542]: what happens when a user doesn't have 1.58 of vscode?

File: packages/plugin-core/src/_extension.ts [hunk 2]
```diff
@@ -522,4 +524,55 @@ async function showWelcomeOrWhatsNew({
...
+function shouldDisplayLapsedUserMsg(): boolean {
+  const ONE_WEEK = Duration.fromObject({ weeks: 1 });
+  const CUR_TIME = Duration.fromObject({ seconds: Time.now().toSeconds() });
+  const metaData = MetadataService.instance().getMeta();
+
+  // If we haven't prompted the user yet and it's been a week since their
+  // initial install OR if it's been one week since we last prompted the user
+  const refreshMsg =
+    (!metaData.lapsedUserMsgSendTime &&
+      ONE_WEEK <=
```
Comment (kevinslin) [inline packages/plugin-core/src/_extension.ts:566]: for first install i would shorten this to 1 day

File: packages/plugin-core/src/_extension.ts [hunk 3]
```diff
@@ -522,4 +524,55 @@ async function showWelcomeOrWhatsNew({
...
+        const ws = getWS();
+        return ws.showWelcome();
+      } else {
+        AnalyticsUtils.track(VSCodeEvents.LapsedUserMessageRejected);
+        return;
+      }
+    });
+}
+
+function shouldDisplayLapsedUserMsg(): boolean {
```
Comment (kevinslin) [inline packages/plugin-core/src/_extension.ts:557]: test please :)

**Review by kevinslin — CHANGES_REQUESTED** [review]

File: packages/plugin-core/src/_extension.ts (hunk 1 above)
Comment (jonathanyeung) [reply packages/plugin-core/src/_extension.ts:542]: Actually, it turns out this feature came out long before 1.58.  The only feature added in 1.58 is the ability to add a `detail` prop in the options, which I decided not to utilize.

File: packages/plugin-core/src/_extension.ts (hunk 3 above)
Comment (jonathanyeung) [reply packages/plugin-core/src/_extension.ts]: tests added!

**Review by kevinslin — APPROVED** [review]
